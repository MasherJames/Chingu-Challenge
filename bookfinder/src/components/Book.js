import React from "react";
import NoPic from "../assets/img/i1.png";

const Book = ({ books }) => {
  const authrs = authors => (authors ? authors : "N/A");
  const imgLink = imageLinks =>
    imageLinks ? imageLinks.smallThumbnail : NoPic;
  const pblisher = publisher => (publisher ? publisher : "N/A");
  const ttle = title => (title ? title : "N/A");
  const inflink = infolink => (infolink ? infolink : "url");

  return books.map(book => {
    const {
      id,
      volumeInfo: { imageLinks, publisher, authors, title, infolink }
    } = book;

    return (
      <div key={id}>
        <img src={imgLink(imageLinks)} alt={ttle(title)} />
        <div>Title: {ttle(title)}</div>
        <div>publisher: {pblisher(publisher)}</div>
        <div>authors: {authrs(authors)}</div>
        <a href={inflink(infolink)}>book info</a>
      </div>
    );
  });
};

export default Book;
