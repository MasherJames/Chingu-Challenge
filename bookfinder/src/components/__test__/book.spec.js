import EnzymeAdapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import Book from "../Book";
import React from "react";

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Book test", () => {
  let defProps = [
    {
      id: "a",
      volumeInfo: {
        imageLinks: "k",
        publisher: "l",
        authors: "z",
        title: "y",
        infolink: "b"
      }
    },
    {
      id: "b",
      volumeInfo: {
        imageLinks: "k",
        publisher: "l",
        authors: "z",
        title: "y",
        infolink: "b"
      }
    },
    {
      id: "c",
      volumeInfo: {
        imageLinks: "k",
        publisher: "l",
        authors: "z",
        title: "y",
        infolink: "b"
      }
    }
  ];

  it("renders correctly", () => {
    const wrapper = shallow(<Book books={defProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
