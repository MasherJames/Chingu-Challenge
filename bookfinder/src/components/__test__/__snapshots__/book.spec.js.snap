// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book test renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Book
    books={
      Array [
        Object {
          "id": "a",
          "volumeInfo": Object {
            "authors": "z",
            "imageLinks": "k",
            "infolink": "b",
            "publisher": "l",
            "title": "y",
          },
        },
        Object {
          "id": "b",
          "volumeInfo": Object {
            "authors": "z",
            "imageLinks": "k",
            "infolink": "b",
            "publisher": "l",
            "title": "y",
          },
        },
        Object {
          "id": "c",
          "volumeInfo": Object {
            "authors": "z",
            "imageLinks": "k",
            "infolink": "b",
            "publisher": "l",
            "title": "y",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "a",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="y"
        />,
        <div>
          Title: 
          y
        </div>,
        <div>
          publisher: 
          l
        </div>,
        <div>
          authors: 
          z
        </div>,
        <a
          href="b"
        >
          book info
        </a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "y",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Title: ",
            "y",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Title: ",
          "y",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "publisher: ",
            "l",
          ],
        },
        "ref": null,
        "rendered": Array [
          "publisher: ",
          "l",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "authors: ",
            "z",
          ],
        },
        "ref": null,
        "rendered": Array [
          "authors: ",
          "z",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "book info",
          "href": "b",
        },
        "ref": null,
        "rendered": "book info",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "a",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="y"
          />,
          <div>
            Title: 
            y
          </div>,
          <div>
            publisher: 
            l
          </div>,
          <div>
            authors: 
            z
          </div>,
          <a
            href="b"
          >
            book info
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "y",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "y",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "y",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "publisher: ",
              "l",
            ],
          },
          "ref": null,
          "rendered": Array [
            "publisher: ",
            "l",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "authors: ",
              "z",
            ],
          },
          "ref": null,
          "rendered": Array [
            "authors: ",
            "z",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "book info",
            "href": "b",
          },
          "ref": null,
          "rendered": "book info",
          "type": "a",
        },
      ],
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "b",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="y"
          />,
          <div>
            Title: 
            y
          </div>,
          <div>
            publisher: 
            l
          </div>,
          <div>
            authors: 
            z
          </div>,
          <a
            href="b"
          >
            book info
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "y",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "y",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "y",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "publisher: ",
              "l",
            ],
          },
          "ref": null,
          "rendered": Array [
            "publisher: ",
            "l",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "authors: ",
              "z",
            ],
          },
          "ref": null,
          "rendered": Array [
            "authors: ",
            "z",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "book info",
            "href": "b",
          },
          "ref": null,
          "rendered": "book info",
          "type": "a",
        },
      ],
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "c",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="y"
          />,
          <div>
            Title: 
            y
          </div>,
          <div>
            publisher: 
            l
          </div>,
          <div>
            authors: 
            z
          </div>,
          <a
            href="b"
          >
            book info
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "y",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "y",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "y",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "publisher: ",
              "l",
            ],
          },
          "ref": null,
          "rendered": Array [
            "publisher: ",
            "l",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "authors: ",
              "z",
            ],
          },
          "ref": null,
          "rendered": Array [
            "authors: ",
            "z",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "book info",
            "href": "b",
          },
          "ref": null,
          "rendered": "book info",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
